theopensystemslab/planx-new
# feat: Invite to pay (applicant/agent journey)

# What does this PR do?
- Allows applicants to generate a `payment_request` record, and directs them to a confirmation page

## Context
The commits might make better sense of this, but there's a bit of a refactor to split the `InviteToPayForm` out of the `Confirm` component, and to use Formik to handle form state as things grew more complex.

## Next steps (upcoming PRs)
 - E2E tests of routing and generation of `payment_request` records
 - Create `TeamStore` to enable more dynamic content - we currently already fetch this data in the `StandaloneView` but aren't able to utilise it

## Testing
[Example flow on Pizza](https://1603.planx.pizza/buckinghamshire/daf-pay-test/preview?analytics=false)
 - Set `INVITE_TO_PAY` and `ALT_THEME` feature flags
 - Take an invite to pay journey, get directed to confirmation page
 - Check record in Hasura

![image](https://user-images.githubusercontent.com/20502206/231228173-05b4d893-1bab-42b8-a07e-8de9da908233.png)
